provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -480,6 +484,40 @@ func (c *Client) addNetworkDevice(
 	return &networkDevice, nil
 }
 
+// GenerateMAC generates a random hardware address that meets VMWare
+// requirements for MAC address: 00:50:56:XX:YY:ZZ where XX is between 00 and 3f.
+// https://pubs.vmware.com/vsphere-4-esx-vcenter/index.jsp?topic=/com.vmware.vsphere.server_configclassic.doc_41/esx_server_config/advanced_networking/c_setting_up_mac_addresses.html
+func GenerateMAC() (string, error) {
+	c, err := rand.Int(rand.Reader, big.NewInt(4194303))
+	if err != nil {
+		return "", err
+	}
+	r := c.Uint64()
+	return fmt.Sprintf("00:50:56:%.2x:%.2x:%.2x", (r>>16)&0xff, (r>>8)&0xff, r&0xff), nil
```

Comment:
00:50:56 is really a const specific to vmware. Can we define and use it instead everywhere?

Reply:
I don't understand - it's currently used only in GenerateMAC and VerifyMAC, and in two very different forms. How would you like to 'unify' it?